pkg/auth/oidc.go

```diff
@@ -0,0 +1,109 @@
+package auth
+
+import (
+	"context"
+	"strings"
+
+	"github.com/grpc-ecosystem/go-grpc-middleware/logging/logrus/ctxlogrus"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/api/idtoken"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+)
+
+var InvalidAuthError = status.Errorf(codes.Unauthenticated, "invalid authentication credentials")
+
+type Validator interface {
+	Validate(ctx context.Context, token string, audience string) (*idtoken.Payload, error)
+}
+
+type OidcValidator struct {
+	validator         Validator
+	audience          string
+	validEmailDomains []string
+	headerKey         string
+	contextKey        interface{}
+}
+
+type OidcValidatorParams struct {
+	Audience          string
+	ValidEmailDomains string
+	HeaderKey         string
+	ContextKey        interface{}
+}
+
+func NewOidcValidator(validator Validator, config *OidcValidatorParams) *OidcValidator {
+	audience := config.Audience
```

Comment:
this will throw panic if config is nil. is it necessary to pass config as pointer?

Reply:
nice catch, I have modified it into non-pointer one to force user to call it by non-nil params